cfpb/consumerfinance.gov
# Lowercases all ireg links

Lowercases all ireg links and also allows for lowercase routable links by filtering on case insensitive when building query set for section.

Fixes https://ghe/CFGOV/platform/issues/3852


## Additions

`lower()` in cfgov/regulations3k/views.py and models/pages.py Python code
- interp-{appendix.lower()}
- interp-{section.lower()}
- section_label.lower()


## Changes

- `Interp -> interp` in `test_resolver.py` and `test_views.py`


## How to test this PR
To check for formatting errors
`tox -e lint`

To run all tests:
`tox -e unittest-current`

To only run regulations3k tests:
1. tox -e unittest-current regulations3k.tests.test_resolver.ReferenceResolutionTestCase
2. tox -e unittest-current regulations3k.tests.test_views.RedirectRegulations3kTestCase

## Screenshots


## Notes and todos

-


## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
